_docs-sources/pipelines/architecture/index.md

```diff
@@ -0,0 +1,45 @@
+# Architecture
+
+## Overview
+
+There are three major components in the Gruntwork Pipelines architecture:
+
+1. The orchestrator
+2. The dispatcher
+3. The executor
+
+### The orchestrator
+
+The orchestrator identifies each [infrastructure change](../core-concepts/#infrastructure-change) in the Pull Request or git commit, classifies the type of infrastructure change it is (e.g. NewAwsAccount vs. OpenTfModuleUpdate) and determines the right [pipelines actions](../core-concepts/#pipelines-actions) to run based on that infrastructure change. For each infrastructure change, it calls the dispatcher.
+
+### The dispatcher
+
+The dispatcher takes as input an infrastructure change and a list of which pipelines actions to run with along with what parameters, and calls the executor to run these actions on the right infrastructure change.
```

Comment:
Currently it only takes a single action.

Reply:
> Also, plan and apply/destroy are two separate actions!

I think that's a bit misleading - we only support one _action_ per infra change. So we're going to plan or apply/destroy. I'll think of some phrasing.

Reply:
Oh, maybe you could help me build an accurate mental model there. In my mind an infra-change is "proposed" in a PR, then we run the first action on it (plan). Then when it's merged (for now, later we'll do some other trigger for applies), we run the apply/destroy action. How do you think about it?

Reply:
You're correct in the sense that a single infra change can have multiple actions run on it as a result of separate GitHub events (e.g., a PR open/sync or merging a PR to main). 

The point I was hoping to make here is that per invocation of pipelines there is only one action (either a plan, apply, or destroy) per infra change.

Reply:
Oh, I see. That makes sense. 👍 

I suppose in the future, a single git event could still trigger more than one action (e.g. plan, then cost estimate), but yes, for now, each "run" supports one pipelines action only.